include/soci/type-conversion.h

```diff
@@ -28,7 +28,12 @@ namespace details
 template <typename T>
 struct base_value_holder
 {
+    base_value_holder()
```

Comment:
That won't change anything here, since number of template instantiations will be same after moving this one member to a base non-template class, and there is no functionality to be moved to such non-template base - it is only for data storage.